packages/admin/admin/src/table/paging/createOffsetLimitPagingAction.ts

```diff
@@ -10,8 +10,9 @@ export function createOffsetLimitPagingAction<TData extends OffsetLimitPagingDat
     { totalCount }: TData,
     limit: number,
 ): IPagingInfo {
-    const currentPage = pagingApi.currentPage ?? 1;
     const totalPages = Math.ceil(totalCount / limit);
+    const currentPage = Math.floor(pagingApi.current / limit) + 1;
+    pagingApi.changePage(currentPage);
```

Comment:
good question

Reply:
To sync the state of the `pagingApi` with the actual application state.

https://github.com/vivid-planet/comet/blob/e10ad7533df3613ee9523896a975b8e95acc9032/packages/admin/admin/src/table/useTableQueryPaging.tsx#L5-L11

The `pagingApi` returns the currentPage. If the `currentPage` is recalculated inside `createOffsetLimitPagingAction.ts`, the `currentPage` returned by the `pagingApi` is incorrect.

The sync is to avoid bugs caused by the developer using the incorrect `currentPage` from `pagingApi`.